rewrite improving-the-documentation.adoc

djencks opened the pull request:
I think this is a more realistic guide to how to work on the documentation than what it replaces.  It also is an experiment about a simpler way to get a website PR/preview build.

davsclaus commented:
[ERROR] Unresolved xrefs:
[ERROR]   manual:ROOT:improving-the-documentation.adoc (/Users/davsclaus/workspace/camel/docs/user-manual/modules/ROOT/pages/improving-the-documentation.adoc) at line 207: xref:eips:enterprise-integration-patterns.adoc[]
[ERROR]   manual:ROOT:improving-the-documentation.adoc (/Users/davsclaus/workspace/camel/docs/user-manual/modules/ROOT/pages/improving-the-documentation.adoc) at line 212: xref:next@components:eips:enterprise-integration-patterns.adoc[]
[ERROR]   manual:ROOT:improving-the-documentation.adoc (/Users/davsclaus/workspace/camel/docs/user-manual/modules/ROOT/pages/improving-the-documentation.adoc) at line 227: xref:user-manual::improving-the-documentation.adoc[]
[ERROR]   manual:ROOT:improving-the-documentation.adoc (/Users/davsclaus/workspace/camel/docs/user-manual/modules/ROOT/pages/improving-the-documentation.adoc) at line 260: xref:{camel-version}@camel:eips:enterprise-integration-patterns.adoc[]

davsclaus commented:
I think we need to separate this into two pages / sections. One for a beginner that just want to help fixing a typo, adding a paragraph to an existing page, etc. vs Camel comitters that maintain several sub projects and needs to better understand how these are linked together, and how it works with antora etc.

The docs may be too overwhelming for the beginner that just want an easy way to contribute - eg browse the docs, click the "edit this page" in the bottom, edit/preview on github, and click submit as PR. That we absolutely need to tell so we can have new users easily contribute.

oscerd commented:
[ERROR] Unresolved xrefs:
[ERROR]   manual:ROOT:improving-the-documentation.adoc (/home/oscerd/workspace/apache-camel/camel/docs/user-manual/modules/ROOT/pages/improving-the-documentation.adoc) at line 207: xref:eips:enterprise-integration-patterns.adoc[]
[ERROR]   manual:ROOT:improving-the-documentation.adoc (/home/oscerd/workspace/apache-camel/camel/docs/user-manual/modules/ROOT/pages/improving-the-documentation.adoc) at line 212: xref:next@components:eips:enterprise-integration-patterns.adoc[]
[ERROR]   manual:ROOT:improving-the-documentation.adoc (/home/oscerd/workspace/apache-camel/camel/docs/user-manual/modules/ROOT/pages/improving-the-documentation.adoc) at line 227: xref:user-manual::improving-the-documentation.adoc[]
[ERROR]   manual:ROOT:improving-the-documentation.adoc (/home/oscerd/workspace/apache-camel/camel/docs/user-manual/modules/ROOT/pages/improving-the-documentation.adoc) at line 260: xref:{camel-version}@camel:eips:enterprise-integration-patterns.adoc[]

oscerd commented:
This needs to be fixed before 3.14.0, otherwise the build will fail

djencks commented:
What exactly are you doing to get these errors? They are all in example blocks so the xrefs should not be evaluated, and for me they are not.

The first two xrefs are correct, they are illustrating xrefs from a different component, but the last two happen to have errors.  Still, something is different between my setup and yours, a local website build against the remote repos works fine for me.

> On Dec 1, 2021, at 2:01 AM, Andrea Cosentino ***@***.***> wrote:
> 
> 
> [ERROR] Unresolved xrefs:
> [ERROR] manual:ROOT:improving-the-documentation.adoc (/home/oscerd/workspace/apache-camel/camel/docs/user-manual/modules/ROOT/pages/improving-the-documentation.adoc) at line 207: xref:eips:enterprise-integration-patterns.adoc[]
> [ERROR] manual:ROOT:improving-the-documentation.adoc (/home/oscerd/workspace/apache-camel/camel/docs/user-manual/modules/ROOT/pages/improving-the-documentation.adoc) at line 212: ***@***.***:eips:enterprise-integration-patterns.adoc[]
> [ERROR] manual:ROOT:improving-the-documentation.adoc (/home/oscerd/workspace/apache-camel/camel/docs/user-manual/modules/ROOT/pages/improving-the-documentation.adoc) at line 227: xref:user-manual::improving-the-documentation.adoc[]
> [ERROR] manual:ROOT:improving-the-documentation.adoc (/home/oscerd/workspace/apache-camel/camel/docs/user-manual/modules/ROOT/pages/improving-the-documentation.adoc) at line 260: ***@***.*** <https://github.com/Camel>:eips:enterprise-integration-patterns.adoc[]
> 
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub <https://github.com/apache/camel/pull/6485#issuecomment-983477439>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAELDXSFP4DUFVAZNK234BTUOXW6PANCNFSM5I4TDTDA>.
>

djencks commented:
There are already two pages explaining how to work on the website.

Currently the faq page manual/faq/how-do-i-edit-the-website.html explains in perhaps appropriate detail how to use the “edit this page” button.  I added some notes on when not to try to use it.  It used to have a really incomplete description of how to do a local build: I removed that and pointed to the other page,

manual/improving-the-documentation.html which now has a note mentioning when using the “edit this page” is appropriate, and more complete instructions on more complex edits.

I did’t want to change things too much :-)

Without the constraints of history, what I’d like is to have all the instructions on the manual/improving-the-documentation.html page in  top-level sections, and a little bit of explanation with two links but no instructions on the faq page.  Shall I make that change and we can discuss the result?

> On Nov 30, 2021, at 10:11 PM, Claus Ibsen ***@***.***> wrote:
> 
> 
> I think we need to separate this into two pages / sections. One for a beginner that just want to help fixing a typo, adding a paragraph to an existing page, etc. vs Camel comitters that maintain several sub projects and needs to better understand how these are linked together, and how it works with antora etc.
> 
> The docs may be too overwhelming for the beginner that just want an easy way to contribute - eg browse the docs, click the "edit this page" in the bottom, edit/preview on github, and click submit as PR. That we absolutely need to tell so we can have new users easily contribute.
> 
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub <https://github.com/apache/camel/pull/6485#issuecomment-983321551>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAELDXWZZYRIW56CXQLD3WDUOW37NANCNFSM5I4TDTDA>.
>

oscerd commented:
> What exactly are you doing to get these errors? They are all in example blocks so the xrefs should not be evaluated, and for me they are not. The first two xrefs are correct, they are illustrating xrefs from a different component, but the last two happen to have errors. Still, something is different between my setup and yours, a local website build against the remote repos works fine for me.
> On Dec 1, 2021, at 2:01 AM, Andrea Cosentino ***@***.***> wrote: [ERROR] Unresolved xrefs: [ERROR] manual:ROOT:improving-the-documentation.adoc (/home/oscerd/workspace/apache-camel/camel/docs/user-manual/modules/ROOT/pages/improving-the-documentation.adoc) at line 207: xref:eips:enterprise-integration-patterns.adoc[] [ERROR] manual:ROOT:improving-the-documentation.adoc (/home/oscerd/workspace/apache-camel/camel/docs/user-manual/modules/ROOT/pages/improving-the-documentation.adoc) at line 212: ***@***.***:eips:enterprise-integration-patterns.adoc[] [ERROR] manual:ROOT:improving-the-documentation.adoc (/home/oscerd/workspace/apache-camel/camel/docs/user-manual/modules/ROOT/pages/improving-the-documentation.adoc) at line 227: xref:user-manual::improving-the-documentation.adoc[] [ERROR] manual:ROOT:improving-the-documentation.adoc (/home/oscerd/workspace/apache-camel/camel/docs/user-manual/modules/ROOT/pages/improving-the-documentation.adoc) at line 260: ***@***.*** <[https://github.com/Camel>:eips:enterprise-integration-patterns.adoc[]](https://github.com/Camel>:eips:enterprise-integration-patterns.adoc%5B%5D) — You are receiving this because you modified the open/close state. Reply to this email directly, view it on GitHub <[#6485 (comment)](https://github.com/apache/camel/pull/6485#issuecomment-983477439)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAELDXSFP4DUFVAZNK234BTUOXW6PANCNFSM5I4TDTDA>.
> Plus... I triggered the jenkins build, which completed without errors: https://ci-builds.apache.org/job/Camel/job/Camel.website/job/main/575/

We are compiling camel. Look at the docs module. Maybe you should build camel when changing this stuff.

oscerd commented:
https://github.com/apache/camel/blob/main/docs/pom.xml#L102

oscerd commented:
https://github.com/apache/camel/blob/main/tooling/maven/camel-package-maven-plugin/src/main/java/org/apache/camel/maven/packaging/XRefCheckMojo.java

If this is not updated/not aligned, then we can simply skip the execution and remove it. If not, we need to fix the errors.

djencks commented:
This is Guillaume's attempt to imitate Antora? I'm afraid this is proof that it doesn't work and needs to be removed.   I can make a PR tomorrow.

davsclaus commented:
No it should not be removed, it catches a number of xref errors when we work on the project.

Instead just update the adoc file you modified and remove links etc - its not an important doc for end users

djencks commented:
There have been a few doc changes more or less recently that moved or removed pages and broke the website build. Were they relying on this xref checker for validation? If so I think that's an additional argument for removing the xref checker.

oscerd commented:
It could be, but it's important to check the build. If you don't want to build locally at least take a look at Jenkins. I'll remove that mojo btw

oscerd commented:
So do we want to maintain the xref checker? It's true that it will show the errors, but i don't think the build should fail

djencks commented:
I don’t think the xref checker is a good idea. It catches some xref problems within the user manual and components components but has false positives due to incomplete understanding of AsciiDoc and completely misses problems with links into these components. I think if you change the docs you need to build the website to find out if you broke anything.

Sent from my iPhone

> On Dec 1, 2021, at 10:29 PM, Andrea Cosentino ***@***.***> wrote:
> 
> ﻿
> So do we want to maintain the xref checker? It's true that it will show the errors, but i don't think the build should fail
> 
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub, or unsubscribe.

oscerd commented:
I know I have to build the website, my question was related to the mojo. If the mojo is completely out of the game, we should remove it. If you modify something on Camel, you NEED to check the camel build not the website build. Website is important, a Camel green build is important too, so please, next time, check the Camel build.

djencks commented:
If you really want to keep the xref checker I can probably insert some invisible stuff into the problematic xrefs so they aren’t recognized by the checker.

Sent from my iPhone

> On Dec 1, 2021, at 10:29 PM, Andrea Cosentino ***@***.***> wrote:
> 
> ﻿
> So do we want to maintain the xref checker? It's true that it will show the errors, but i don't think the build should fail
> 
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub, or unsubscribe.

oscerd commented:
If it's giving false positives, I don't see the value of having it. We are changing stuff around the Antora content and nobody cared about this Mojo. Maybe we need to put more attention on this stuff. There's not only Antora and the website in this ecosystem.

djencks commented:
I think I did check the PR builds for this commit, and they had failed for an unrelated check style problem. The ones I can find now certainly did. Since I’d checked the complete website build I figured it was ok.

Sent from my iPhone

> On Dec 1, 2021, at 10:23 PM, Claus Ibsen ***@***.***> wrote:
> 
> ﻿
> No it should not be removed, it catches a number of xref errors when we work on the project.
> 
> Instead just update the adoc file you modified and remove links etc - its not an important doc for end users
> 
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub, or unsubscribe.

oscerd commented:
Yes, but the camel build is failing.

davsclaus commented:
For example having this in docs in the core camel project, such as in bean-eip.adoc

xref:does-not-exists.adoc[XXX] bla blah

Then the validator catches this

[INFO] --- camel-package-maven-plugin:3.14.0-SNAPSHOT:xref-check (xref-check) @ docs ---
[ERROR] Unresolved xrefs:
[ERROR]   components:eips:bean-eip.adoc (/Users/davsclaus/workspace/camel/core/camel-core-engine/src/main/docs/modules/eips/pages/bean-eip.adoc) at line 14: xref:does-not-exists.adoc[XXX]

I am going to comment out / change that adoc file so the validator works and the CI builds works again - we need this for building SNAPSHOTS and to release 3.14.x

Those 4 xref failures are not useable anyway as linking to EIPs, user manual etc can easily find without a link, and this doc page is not for normal Camel users.

davsclaus commented:
Okay did a "clever hack" and use XREF in upper case for that "code output". I do not know if there is some escape char you can use, so you can try to escape xref so its \x\r\e\f or something but still displayed as xref in lower case

djencks commented:
See https://github.com/apache/camel/pull/6497 for better ugly hack.
